Add more double/duplicate operations tests

hwwhww opened the pull request:
Thanks @ajsutton for proposing this new test case.

- **[no substantive  change]** move `test_exit_and_bls_change` to `test_success_exit_and_bls_change`
- add `test_invalid_double_bls_changes_same_block`
    - a block includes the same `BLSToExecutionChange` of the same validator. The first `BLSToExecutionChange` will succeed but the second will fail due to the assertion `assert validator.withdrawal_credentials[:1] == BLS_WITHDRAWAL_PREFIX`.

djrtwo reviewed (approved):
looks good!

djrtwo commented on tests/core/pyspec/eth2spec/test/capella/sanity/test_blocks.py:
```diff
@@ -44,6 +48,58 @@ def test_successful_bls_change(spec, state):
+
+    # Double BLSToExecutionChange of the same validator
+    for _ in range(2):
+        block.body.bls_to_execution_changes.append(signed_address_change)
```
for completeness, let's have two tests
1. of the same exact bls change
2. of two valid bls changes for the same validator but to a different 0x01 address

hwwhww replied:
added `test_invalid_two_bls_changes_of_different_addresses_same_validator_same_block`

hwwhww commented:
reviewed and added new double operation tests:
- existing old tests:
    - `test_invalid_double_same_proposer_slashings_same_block`
    - `test_invalid_double_same_attester_slashing_same_block`
    - `test_invalid_double_validator_exit_same_block`
- added new tests in this PR:
    - `test_double_same_attestation_same_block` note: it's **valid**
    - `test_invalid_double_same_deposit_same_block`

djrtwo commented on tests/core/pyspec/eth2spec/test/capella/sanity/test_blocks.py:
```diff
@@ -44,6 +48,82 @@ def test_successful_bls_change(spec, state):
+
+@with_phases([CAPELLA])
+@spec_state_test
+def test_invalid_double_bls_changes_same_block(spec, state):
```
`duplicate` might be a bit more accurate name for *precisely* the same message

djrtwo commented on tests/core/pyspec/eth2spec/test/phase0/sanity/test_blocks.py:
```diff
@@ -744,6 +744,27 @@ def test_deposit_in_block(spec, state):
 
+@with_all_phases
+@spec_state_test
+def test_invalid_double_same_deposit_same_block(spec, state):
```
duplicate means "double same" essentially
```suggestion
def test_invalid_duplicate_deposit_same_block(spec, state):
```

djrtwo reviewed (approved):
great!
